provider/azure/environ.go

```diff
@@ -1243,17 +1260,33 @@ func (env *azureEnviron) deleteVirtualMachine(
 
 	// The VM must be deleted first, to release the lock on its resources.
 	logger.Debugf("- deleting virtual machine (%s)", vmName)
-	vmResultCh, errCh := vmClient.Delete(env.resourceGroup, vmName, nil)
-	if result, err := <-vmResultCh, <-errCh; err != nil {
-		msg := "deleting virtual machine"
+	vmErrMsg := "deleting virtual machine"
+	vmFuture, err := vmClient.Delete(sdkCtx, env.resourceGroup, vmName)
+	if err != nil {
 		if errorutils.MaybeInvalidateCredential(err, ctx) {
-			return invalidatedCredential(err, msg)
+			return invalidatedCredential(err, vmErrMsg)
 		}
-		if !isNotFoundResponse(result.Response) {
-			return errors.Annotate(err, msg)
+		if !isNotFoundResponse(vmFuture.Response()) {
+			return errors.Annotate(err, vmErrMsg)
+		}
+	} else {
+		err = vmFuture.WaitForCompletionRef(sdkCtx, vmClient.Client)
+		if err != nil {
+			if errorutils.MaybeInvalidateCredential(err, ctx) {
```

Comment:
So here is where it really doesn't make sense.
You just need
return errorutils.HandleCredentialError(errors.Annotate(err, "deleting virtual machine"), ctx)